packages/files/utils.js

```diff
@@ -0,0 +1,388 @@
+                    sendError(res, 413, `The ‘${fieldName}’ multipart field value exceeds the ${processRequestOptions.maxFieldSize} byte size limit.`)
+                )
+            }
+
+            if (fieldName === 'meta') {
+                meta = parseAndValidateMeta(value, req, res, exit)
+            }
+
+            req.meta = meta
+        })
+
+        parser.once('finish', () => {
+            req.unpipe(parser)
+            req.resume()
+
+            if (!Array.isArray(req.files)) {
+                return exit(() => sendError(res, 400, 'Missing attached files'))
+            }
+            if (!meta) {
+                return exit(() => sendError(res, 400, 'Missing multipart field "meta"'))
+            }
+            next()
+        })
+
+        req.once('close', () => {
+            if (!req.readableEnded) {
+                exit(() => sendError(res, 499, 'Request disconnected during file upload stream parsing.'))
+            }
+        })
+
+        res.once('close', () => { released = true })
+        req.pipe(parser)
+    }
+}
+
+const fileStorageHandler = ({ keystone, appClients }) => {
+    return async function (req, res) {
+        const { meta, files } = req
+        const appClient = appClients ? appClients[meta.appId] : undefined
+
+        if (!(meta['appId'] in (appClients || {}))) {
+            return res.status(403).json({ error: `${meta['appId']} does not have permission to upload files` })
+        }
+
+        const fileAdapter = new FileAdapter(meta['appId'])
+        const context = keystone.createContext({ skipAccessControl: true })
+
+        const savedFiles = await Promise.all(
+            files.map(file =>
+                fileAdapter.save({
+                    stream: file.createReadStream(),
+                    filename: file.filename,
+                    mimetype: file.mimetype,
+                    encoding: file.encoding,
+                    id: cuid(),
+                    meta,
+                })
+            )
+        )
+
+        const condoFiles = await CondoFile.createMany(
+            context,
+            savedFiles.map((data, index) => {
+                const fileData = {
+                    ...data,
+                    originalFilename: files[index].filename,
+                    mimetype: files[index].mimetype,
+                    encoding: files[index].encoding,
+                    meta,
+                }
+
+                const signature = jwt.sign(fileData, appClient.secret, { expiresIn: '5m' })
+
+                return {
+                    data: {
+                        file: fileData,
+                        dv: meta.dv,
+                        sender: meta.sender,
+                        signature,
+                        user: { connect: { id: req.user.id } },
+                    },
+                }
+            }),
+            'signature file { id }'
+        )
+
+        res.json(condoFiles)
+    }
+}
+
+const attachHandler = ({ keystone }) => {
```

Comment:
We need to think about this. 

Our clients might not use this endpoint, and we does not notify them in current implementation, what file can be deleted

Reply:
Yep, i'll remove it